### PR TITLE
[Docs] improve job resources page

### PR DESCRIPTION
**Pull request by jillnogold**

### CODE REVIEW

File: docs/runtimes/configuring-job-resources.md [hunk 1]
```diff
@@ -37,73 +37,72 @@ fn.set_envs(file_path="env.txt")
...
+training_function.spec.replicas = 2
 ```
-spec.min_replicas = 1
-spec.max_replicas = 4
+or a range (for auto-scaling in Dask or Nuclio):
+```python
+training_function = mlrun.code_to_function("training.py", name="training", handler="train", 
+                                           kind="mpijob", image="mlrun/mlrun-gpu")
+training_function.min_replicas = 1
+training_function.max_replicas = 4
```
Comment (alonmr) [inline docs/runtimes/configuring-job-resources.md]: this says dask or nuclio but kind is mpijob

File: docs/runtimes/configuring-job-resources.md [hunk 2]
```diff
@@ -207,22 +208,21 @@ Preemption mode has three values:
...
+# Only run on non-spot instances
+fn.with_node_selection(node_selector={"app.iguazio.com/lifecycle" : "non-preemptible"})
 ```
+
+And another function that can only be scheduled on preemptible noodes:
+
+
+```python
 import mlrun
 import os
```
Comment (alonmr) [inline docs/runtimes/configuring-job-resources.md]: remove this line

File: docs/runtimes/configuring-job-resources.md [hunk 3]
```diff
@@ -207,22 +208,21 @@ Preemption mode has three values:
...
-
 ### SDK configuration
 
 Configure preemption mode by adding the `with_preemption_mode` parameter in your Jupyter notebook, and specifying a mode from the list of values above. <br>
 This example illustrates a function that cannot be scheduled on preemptible nodes:
 
+
+```python
+# Only run on non-spot instances
+fn.with_node_selection(node_selector={"app.iguazio.com/lifecycle" : "non-preemptible"})
```
Comment (alonmr) [inline docs/runtimes/configuring-job-resources.md]: this looks like its misplaced no? it also a few lines below

File: docs/runtimes/configuring-job-resources.md [hunk 4]
```diff
@@ -234,11 +234,13 @@ train_fn.run(inputs={"dataset": my_data})
...
-Alternatively, you can specify the preemption using `with_priority_class` and `with_node_selection` parameters. This example specifies that the pod/function runs only on non-preemptible nodes:
+See {py:meth}`~#RemoteRuntime.with_preemption_mode.
 
-```
+Alternatively, you can specify the preemption using `with_priority_class` and `with_node_selection` parameters. This example specifies that 
+the pod/function runs only on non-preemptible nodes:
+
+```python
 import mlrun
 import os
```
Comment (alonmr) [inline docs/runtimes/configuring-job-resources.md]: remove line not needed

File: docs/runtimes/configuring-job-resources.md [hunk 5]
```diff
@@ -252,45 +254,36 @@ fn.with_node_selection(node_selector={"app.iguazio.com/lifecycle":"non-preemptib
...
 
 ### SDK configuration
 
 Configure pod priority by adding the priority class parameter in your Jupyter notebook. <br>
 For example:
 
-```
+```python
 import mlrun
 import os
```
Comment (alonmr) [inline docs/runtimes/configuring-job-resources.md]: same

File: docs/runtimes/configuring-job-resources.md [hunk 6]
```diff
@@ -301,29 +294,57 @@ train_fn.run(inputs={"dataset" :my_data})
...
-```python
-# Only run on non-spot instances
-fn.with_node_selection(node_selector={"app.iguazio.com/lifecycle" : "non-preemptible"})
+### SDK configuration
+
+Configure node selection by adding the key:value pairs in your Jupyter notebook formatted as a Python dictionary. <br>
+For example:
+
+```python        
+# Add node selection
```
Comment (alonmr) [inline docs/runtimes/configuring-job-resources.md]: comment is redundant

File: docs/runtimes/configuring-job-resources.md [hunk 7]
```diff
@@ -301,29 +294,57 @@ train_fn.run(inputs={"dataset" :my_data})
...
+func.with_node_selection(node_selector={name})
```
Comment (alonmr) [inline docs/runtimes/configuring-job-resources.md]: the node_selector value is a key, value pair and not just a name

**Review by alonmr — COMMENTED** [review]
Took a quick scan and here are a few comments for starters

File: docs/runtimes/configuring-job-resources.md (hunk 1 above)
Comment (jillnogold) [reply docs/runtimes/configuring-job-resources.md]: changed example

File: docs/runtimes/configuring-job-resources.md [hunk 8]
```diff
@@ -207,21 +208,20 @@ Preemption mode has three values:
...
-
 ### SDK configuration
 
 Configure preemption mode by adding the `with_preemption_mode` parameter in your Jupyter notebook, and specifying a mode from the list of values above. <br>
 This example illustrates a function that cannot be scheduled on preemptible nodes:
 
+
+```
+# Only run on non-spot instances
+fn.with_node_selection(node_selector={"app.iguazio.com/lifecycle" : "non-preemptible"})
```
Comment (alonmr) [inline docs/runtimes/configuring-job-resources.md]: This node selector label only applies for iguazio. I believe you can also use `fn. with_preemption_mode("allow")` instead

File: docs/runtimes/configuring-job-resources.md [hunk 9]
```diff
@@ -252,38 +254,29 @@ fn.with_node_selection(node_selector={"app.iguazio.com/lifecycle":"non-preemptib
...
+
+## Pod priority for user jobs
+
+[Priority classes](https://kubernetes.io/docs/concepts/scheduling-eviction/pod-priority-preemption/) are a mechanism in Kubernetes to 
+control the order in which pods are scheduled and evicted &mdash; to make room for other, higher priority pods. Priorities also affect the pods’ 
+evictions in case the node’s memory is pressured (called Node-pressure Eviction).
 
+Pod priority is relevant for all of the jobs created by MLRun. For Nuclio it applies to the pods of the Nuclio-created functions.
+
+Pod priority is specified through Priority classes, which map to a priority value. The priority values are: High, Medium, Low. The default is Medium.
```
Comment (alonmr) [inline docs/runtimes/configuring-job-resources.md]: > The priority values are: High, Medium, Low. The default is Medium. 

That's not always true. they can use `fn.list_valid_priority_class_names()` and `fn.get_default_priority_class_name()` to get these values.

**Review by alonmr — COMMENTED** [review]
Looks good, small comments

File: docs/runtimes/configuring-job-resources.md [hunk 10]
```diff
@@ -1,4 +1,4 @@
-(configuring-job-resources)=
+ (configuring-job-resources)=
```
Comment (alonmr) [inline docs/runtimes/configuring-job-resources.md]: is the whitespace prefix required?

File: docs/runtimes/configuring-job-resources.md [hunk 11]
```diff
@@ -94,15 +94,15 @@ Configure requests and limits in the service's **Common Parameters** tab and in
...
 
 For each Nuclio or serving function, MLRun creates an HTTP trigger with the default of 1 worker.  When using GPU in remote functions you must ensure 
-that the number of GPUs is equal to the number of workers (or manage the GPU consumption within your code). You can set the [number of GPUs for each pod using the MLRun SDK](#cpu-gpu-and-memory-limits-for-user-jobs).
+that the number of GPUs is equal to the number of workers (or manage the GPU consumption within your code). You can set the [number of GPUs for each pod using the MLRun SDK](#cpu-gpu-and-memory-requests-and-limits-for-user-jobs).
 
 You can change the number of workers after you create the trigger (function object), then you need to 
 redeploy the function. Examples of changing the number of workers:
 
-Using {py:meth}`~mlrun.runtimes.html#mlrun.runtimes.RemoteRuntime.with_http`:</br>
+Using {py:meth}`~mlrun.runtimes.KubeResource.with_http`:</br>
```
Comment (alonmr) [inline docs/runtimes/configuring-job-resources.md]: triggers are only for nuclio runtimes so actually it should be `mlrun.runtimes.nuclio.function.RemoteRuntime`

File: docs/runtimes/configuring-job-resources.md [hunk 12]
```diff
@@ -94,15 +94,15 @@ Configure requests and limits in the service's **Common Parameters** tab and in
...
 `serve.with_http(workers=8, worker_timeout=10)`
 
-Using {py:meth}`~mlrun.runtimes.html#mlrun.runtimes.RemoteRuntime.add_v3io_stream_trigger`:</br>
+Using {py:meth}`~mlrun.runtimes.KubeResource.add_v3io_stream_trigger`:</br>
```
Comment (alonmr) [inline docs/runtimes/configuring-job-resources.md]: same

**Review by alonmr — COMMENTED** [review]
minor

**Review by alonmr — APPROVED** [review]

**Pull request merged**